ipatests/test_webui/test_automember.py

```diff
@@ -267,6 +377,313 @@ def test_rebuild_membership_users(self):
         self.assert_record('dev2')
 
         # Delete group, users and automember rule
-        self.delete('group', [{'pkey': 'devel'}])
-        self.delete('user', [{'pkey': 'dev1'}, {'pkey': 'dev2'}])
-        self.delete('automember', [{'pkey': 'devel'}], facet='searchgroup')
+        self.delete_user_groups('devel')
+        self.delete_users('dev1', 'dev2')
+        self.delete_user_group_rules('devel')
+
+    @screenshot
+    def test_add_multiple_user_group_rules(self):
+        """
+        Test creating and deleting multiple user group rules
+        """
+
+        groups = ['group1', 'group2', 'group3']
+
+        for group in groups:
+            self.add_user_group(group)
+
+        self.add_user_group_rules(*groups)
+        self.delete_user_group_rules(*groups)
+
+    @screenshot
+    def test_add_multiple_host_group_rules(self):
+        """
+        Test creating and deleting multiple host group rules
+        """
+
+        groups = ['group1', 'group2', 'group3']
+
+        for group in groups:
+            self.add_host_group(group)
+
+        self.add_host_group_rules(*groups)
+        self.delete_host_group_rules(*groups)
+
+    @screenshot
+    def test_search_user_group_rule(self):
+        """
+        Test searching user group rules using filter
+        """
+
+        pkey = SEARCH_CASES['name']
+        self.add_user_group(pkey, '')
+        self.add_user_group_rules(pkey)
+        self.navigate_to_record(pkey)
+        self.mod_record(ENTITY, {'mod': [
+            ('textarea', 'description', SEARCH_CASES['description']),
+        ]}, facet='usergrouprule')
+        self.navigate_to_entity(ENTITY, facet='searchgroup')
+
+        for text in SEARCH_CASES['positive']:
+            self.apply_search_filter(text)
+            self.wait(0.5)
```

Comment:
shouldn't it wait for something more specific, e.g. hiding of activity indicator (`wait_for_request()`)? What if the tested machine is slower for some reason and the search takes longer than 0.5s? Same in other occurances below.

Reply:
Fixed